test/functorch/test_aotdispatch.py

```diff
@@ -2305,10 +2305,14 @@ def test_aot_autograd_symbolic_exhaustive(self, device, dtype, op):
                                 # of a tracing tensor with aten._local_scalar_dense.default -
                                 # erroring out! It's likely that this is caused by data-dependent
                                 # control flow or similar.
+    torch.nn.TransformerEncoder,  # DataDependentOutputException: aten.equal compares a mask input
```

Comment:
I saw on the original stacktrace that the failing test was named `test_aot_autograd_module_exhaustive_nn_Transformer_cpu_float32` (https://github.com/pytorch/pytorch/actions/runs/3773547267/jobs/6415150476)

Maybe try putting `torch.nn.Transformer` instead?

Reply:
Do we know what the perf implications are for doing this in the non-causal path. I.e Does adding the torch.equal hurt the performance of a TransformerEncoder model that is currently torch.compilable?

Reply:
Would it not remain compilable, but with a break.  So, I'll assert that we have to choose sides here, and causal is a massive preponderance of use cases?  We might provide a way to suppress this (much like offer users ways to suppress using nested tensors, or the alignment mask correctness test during NT creation)?